File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_worker_clause.cpp
#include <iostream>
#include <stdlib.h>
#include <openacc.h>

#ifndef T1
//T1:loop construct worker clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    const int N = 1000;
    int a[N], b[N], c[N];

    for(int i = 0; i < N; i++){
        a[i] = rand() % 100;
        b[i] = rand() % 100;
    }

    #pragma acc kernels copyin(a[0:N], b[0:N]) copyout(c[0:N])
    {
        #pragma acc loop worker
        for(int i = 0; i < N; i++){
            c[i] = a[i] + b[i];
        }
    }

    for(int i = 0; i < N; i++){
        if(c[i] != a[i] + b[i]){
            err = 1;
            break;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-finetuned/stage2/loop_construct_worker_clause.f90
!T1:loop construct worker clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0

        INTEGER :: a(LOOPCOUNT)
        INTEGER :: b(LOOPCOUNT)
        INTEGER :: c(LOOPCOUNT)
        INTEGER :: d(LOOPCOUNT)
        
        ! Loop initializer
        SEEDDIM(1) = 1
        SEEDDIM(2) = 1
        CALL RANDOM_SEED(PUT=SEEDDIM)
        CALL RANDOM_NUMBER(a)
        CALL RANDOM_NUMBER(b)
        c = 0
        d = 0

        !$acc data copyin(a(1:LOOPCOUNT), b(1:LOOPCOUNT)) copy(c(1:LOOPCOUNT),d(1:LOOPCOUNT))
          !$acc parallel
            !$acc loop worker
            DO i = 1, LOOPCOUNT
              c(i) = a(i) + b(i)
            END DO
            !$acc loop worker
            DO i = 1, LOOPCOUNT
              d(i) = a(i) * b(i)
            END DO
          !$acc end parallel
        !$acc end data
        DO i = 1, LOOPCOUNT
          IF (c(i) .ne. a(i) + b(i)) THEN
            errors = errors + 1
          END IF
          IF (d(i) .ne. a(i) * b(i)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM